media/src/commonMain/kotlin/MediaPlayer.kt

```diff
@@ -0,0 +1,442 @@
+/*
+ Copyright 2023 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.media
+
+import com.splendo.kaluga.media.MediaPlayer.Controls.AwaitPreparation
+import com.splendo.kaluga.media.MediaPlayer.Controls.DisplayError
+import com.splendo.kaluga.media.MediaPlayer.Controls.Pause
+import com.splendo.kaluga.media.MediaPlayer.Controls.Play
+import com.splendo.kaluga.media.MediaPlayer.Controls.Seek
+import com.splendo.kaluga.media.MediaPlayer.Controls.SetLoopMode
+import com.splendo.kaluga.media.MediaPlayer.Controls.SetRate
+import com.splendo.kaluga.media.MediaPlayer.Controls.Stop
+import com.splendo.kaluga.media.MediaPlayer.Controls.Unpause
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.flow.transformLatest
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+import kotlin.js.JsName
+import kotlin.jvm.JvmName
+import kotlin.time.Duration
+
+/**
+ * Plays a [MediaSource]
+ */
+interface MediaPlayer : VolumeController, MediaSurfaceController {
+
+    /**
+     * The controls available to manage playback on a [MediaPlayer]
+     * @property play the [Play] control available
+     * @property pause the [Pause] control available
+     * @property unpause the [Unpause] control available
+     * @property stop the [Stop] control available
+     * @property seek the [Seek] control available
+     * @property setRate the [SetRate] control available
+     * @property setLoopMode the [SetLoopMode] control available
+     * @property awaitPreparation the [AwaitPreparation] control available
+     * @property displayError the [DisplayError] control available
+     */
+    data class Controls(
+        val play: Play? = null,
```

Comment:
`fun play():Boolean`so you can just do `controls.play()` rather than `controls.play?[.invoke]()` seems nicer to me. 

Of course one does not exclude the other, but I wonder what you think.